legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/helperFunctions/helperFunctions.pure

```diff
@@ -107,18 +107,37 @@ function meta::relational::functions::database::setUpData(records:List<String>[*
 
    meta::alloy::service::execution::setUpDataSQLs($records, $database);
 }
-
 // This function is needed because split does not preserve empty strings
 // We work around by inserting a non-empty value, doing a split, and then removing the non-empty value
 function <<access.private>> meta::alloy::service::execution::splitWithEmptyValue(s:String[1]) : String[*]
 {
    let delimForEmptyCsvField = '|EMPTY|';
 
    $s->meta::alloy::service::execution::replaceWithEmptyValue($delimForEmptyCsvField)
-        ->split(',')
+        ->meta::alloy::service::execution::splitByDelimiter(',')
         ->map(csv_val| if($csv_val == $delimForEmptyCsvField, | '' ,| $csv_val));
 }
 
+Class meta::alloy::service::execution::SplitHelper
```

Comment:
SplitHelper -> SplitterState (also make the class private)
arr -> tokens
stringBuilder:String[1] -> currentTokenChars:String[*] 
concatenating is inefficient, maintain chars in an array, and call makeString() on it to get str, before adding to tokens